python/cuspatial/cuspatial/tests/binpreds/test_binpred_test_dispatch.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+from functools import wraps
+
+import pandas as pd
+import pytest
+from binpred_test_dispatch import predicate, simple_test  # noqa: F401
+
+"""Decorator function that xfails a test if an exception is throw
+by the test function. Will be removed when all tests are passing."""
+
+
+def xfail_on_exception(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except Exception as e:
+            pytest.xfail(f"Xfailing due to an exception: {e}")
+
+    return wrapper
+
+
+# In the below file, all failing tests are recorded with visualizations.
+out_file = open("test_binpred_test_dispatch.log", "w")
+
+
+@xfail_on_exception  # TODO: Remove when all tests are passing
```

Comment:
All tests are passing now ;)

Reply:
They are but not in this PR, the full submission comes in 3 PRs: #1085, #1086, and #1064